### PR TITLE
Rename DeepSpeedPlugin to DeepSpeedStrategy

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Part of #10549 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @SeanNaren @awaelchli

### CODE REVIEW

File: tests/plugins/test_cluster_integration.py [hunk 1]
```diff
@@ -55,7 +55,8 @@ def environment_combinations():
 
 
 @pytest.mark.parametrize(
-    "plugin_cls", [DDPPlugin, DDPShardedPlugin, DDP2Plugin, pytest.param(DeepSpeedPlugin, marks=RunIf(deepspeed=True))]
+    "plugin_cls",
+    [DDPPlugin, DDPShardedPlugin, DDP2Plugin, pytest.param(DeepSpeedStrategy, marks=RunIf(deepspeed=True))],
 )
 def test_ranks_available_manual_plugin_selection(plugin_cls):
```
Comment (rohitgr7) [inline tests/plugins/test_cluster_integration.py:61]: maybe need to update the parameter name and test name here ?
plugin -> strategy

File: tests/plugins/test_deepspeed_plugin.py [hunk 2]
```diff
@@ -124,7 +124,7 @@ def deepspeed_zero_config(deepspeed_config):
 
 
 @RunIf(deepspeed=True)
-@pytest.mark.parametrize("plugin", ("deepspeed", DeepSpeedPlugin))
+@pytest.mark.parametrize("plugin", ("deepspeed", DeepSpeedStrategy))
```
Comment (rohitgr7) [inline tests/plugins/test_deepspeed_plugin.py:127]: same here.

File: tests/plugins/test_deepspeed_plugin.py [hunk 3]
```diff
@@ -148,7 +148,7 @@ def test_deepspeed_plugin_env(tmpdir, monkeypatch, deepspeed_config):
     trainer = Trainer(fast_dev_run=True, default_root_dir=tmpdir, strategy="deepspeed")
 
     plugin = trainer.training_type_plugin
-    assert isinstance(plugin, DeepSpeedPlugin)
+    assert isinstance(plugin, DeepSpeedStrategy)
```
Comment (rohitgr7) [inline tests/plugins/test_deepspeed_plugin.py:151]: same here.

File: tests/plugins/test_cluster_integration.py (hunk 1 above)
Comment (awaelchli) [reply tests/plugins/test_cluster_integration.py:61]: That would be too far reaching and should be done for follow up PRs. See [#10548 ](https://github.com/PyTorchLightning/pytorch-lightning/issues/10549#issuecomment-997863968)

**Review by awaelchli — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by carmocca — APPROVED** [review]
